snap/local/runtime-helpers/bin/security-proxy-post-setup.sh

```diff
@@ -11,3 +11,6 @@ export PATH="$SNAP/bin:$PATH"
 # Several config options depend on resources that only exist after proxy 
 # setup. This re-applies the config options logic after deferred startup:
 $SNAP/snap/hooks/configure options --service=security-proxy
+
+# Process the EdgeX >=2.2 snap options
+$SNAP/snap/hooks/configure options --service=secrets-config
```

Comment:
I don't love the fact we use the service name `secrets-config` here, as its not really a service, it's an application. Is there a reason you changed this from `security-proxy`?

Reply:
The new options are grouped per app and here, we are calling the processing of options for the secrets-config app. It should have been `--app=secrets-config` for accuracy but that'd involve changes to the parts that are being deprecated. I will add an issue to take care of this as part of future refactoring. But for now I would leave it as is because this is an internal CLI typically not seen or used by any user.

Reply:
That doesn't really make sense as you're not configuring the secrets-config application, you're configuring the proxy, secrets-config is a helper application.

Reply:
If at some point the proxy supports provisioning of users and TLS certificates via environment variables, then we can use the `apps.secrets-proxy.config` namespace to set those. In the meantime and in the scope of the new scheme, a user has to rely on the secrets-config app to perform the operations during runtime by setting e.g. `snap set edgexfoundry apps.secrets-config.proxy.admin.public-key=xxx`. Note that there is no `config` in the key because this isn't a configuration override.

The app in the suggested `--app=<app>` CLI argument is equivalent to `apps.<app>` namespace for snap options. In other words, as said, we are calling the processing of options for the secrets-config app. For now, only proxy is supported so we don't need to specify that in the CLI.